client/my-sites/site-settings/search.jsx

```diff
@@ -0,0 +1,178 @@
+
+	static propTypes = {
+		isSavingSettings: PropTypes.bool,
+		isRequestingSettings: PropTypes.bool,
+		fields: PropTypes.object,
+	};
+
+	renderInfoLink( url ) {
+		const { translate } = this.props;
+
+		return (
+			<div className="search__info-link-container site-settings__info-link-container">
+				<InfoPopover position="left">
+					<ExternalLink href={ url } icon target="_blank">
+						{ translate( 'Learn more about Search.' ) }
+					</ExternalLink>
+				</InfoPopover>
+			</div>
+		);
+	}
+
+	renderSearchExplanation() {
+		const { translate } = this.props;
+
+		return (
+			<FormSettingExplanation>
+				{ translate( 'Your site is using enhanced search powered by Elasticsearch.' ) }
+			</FormSettingExplanation>
+		);
+	}
+
+	renderWpcomSettings() {
+		return (
+			<div>
+				{ this.renderInfoLink( 'https://support.wordpress.com/search/' ) }
+
+				<div>
+					{ this.renderSearchExplanation() }
+				</div>
+			</div>
+		);
+	}
+
+	renderJetpackSettingsContent() {
+		const {
+			activatingSearchModule,
+			searchModuleActive,
+			translate
+		} = this.props;
+
+		return (
+			<div className="search__module-settings site-settings__child-settings">
+				{
+					activatingSearchModule &&
+					<FormSettingExplanation>
+						{ translate( 'Activating search…' ) }
+					</FormSettingExplanation>
+				}
+
+				{
+					searchModuleActive &&
+					<div>
+						{ this.renderSearchExplanation() }
+					</div>
+				}
+			</div>
+		);
+	}
+
+	renderJetpackSettings() {
+		const {
+			isRequestingSettings,
+			isSavingSettings,
+			siteId,
+			translate
+		} = this.props;
+
+		return (
+			<FormFieldset>
+				{ this.renderInfoLink( 'https://jetpack.com/support/search' ) }
+
+				<JetpackModuleToggle
+					siteId={ siteId }
+					moduleSlug="search"
+					label={ translate( 'Enable site-wide search powered by Elasticsearch' ) }
+					disabled={ isRequestingSettings || isSavingSettings }
+				/>
+
+				{ this.renderJetpackSettingsContent() }
+			</FormFieldset>
+		);
+	}
+
+	render() {
+		const {
+			siteId,
+			siteIsJetpack,
+			enableFeature,
+			translate
+		} = this.props;
+
+		// for now, don't even show upgrade nudge
+		if ( ! enableFeature ) {
+			return null;
+		}
+
+		return (
+			<div>
+				{ siteId && <QueryJetpackConnection siteId={ siteId } /> }
+
+				<SectionHeader label={ translate( 'Search' ) } />
+
+				<Card className="search__card site-settings__traffic-settings">
+					{ siteIsJetpack
+						? this.renderJetpackSettings()
+						: this.renderWpcomSettings()
+					}
+				</Card>
+			</div>
+		);
+	}
+}
+
+export default connect(
+	( state ) => {
+		const site = getSelectedSite( state );
+		const siteId = getSelectedSiteId( state );
+		const isSearchEligible = site && site.plan && hasBusinessPlan( site.plan );
+
+		return {
+			siteId,
+			activatingSearchModule: !! isActivatingJetpackModule( state, siteId, 'search' ),
+			enableFeature: isSearchEligible,
+			site: getSelectedSite( state ),
+			siteSlug: getSelectedSiteSlug( state ),
```

Comment:
We are not using `siteSlug` anywhere, so no need to pass it as a connected prop.